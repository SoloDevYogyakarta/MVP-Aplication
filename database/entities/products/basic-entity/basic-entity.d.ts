import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class ProductBasicEntity extends BaseEntity {
    name?: string;
    status?: number;
    condition?: number;
    shortdesc?: string;
    main_stock?: number;
    reserve_stock?: number;
    user_id?: string;
}
export type ProductBasicCreationAttribute = Optional<ProductBasicEntity, 'id'>;
export interface ProductBasicInstance extends Model<ProductBasicCreationAttribute, ProductBasicEntity>, ProductBasicEntity {
}
export declare const productBasicEntity: import("sequelize").ModelCtor<ProductBasicInstance>;
