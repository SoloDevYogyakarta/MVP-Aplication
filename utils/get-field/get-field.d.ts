import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
