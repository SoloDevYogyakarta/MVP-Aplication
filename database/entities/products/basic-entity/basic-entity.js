"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.productBasicEntity = exports.ProductBasicEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class ProductBasicEntity extends base_entity_1.BaseEntity {
}
exports.ProductBasicEntity = ProductBasicEntity;
exports.productBasicEntity = entity_1.sequelize.define('BASIC', {
    public_id: sequelize_1.DataTypes.STRING,
    name: sequelize_1.DataTypes.STRING,
    status: sequelize_1.DataTypes.INTEGER,
    condition: sequelize_1.DataTypes.INTEGER,
    shortdesc: sequelize_1.DataTypes.STRING,
    main_stock: sequelize_1.DataTypes.INTEGER,
    reserve_stock: sequelize_1.DataTypes.INTEGER,
    user_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'BASIC',
    schema: 'PRODUCTS',
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=basic-entity.js.map