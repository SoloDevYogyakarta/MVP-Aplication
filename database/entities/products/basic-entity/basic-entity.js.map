{"version":3,"file":"basic-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/products/basic-entity/basic-entity.ts"],"names":[],"mappings":";;;AAAA,mCAAgC;AAChC,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,kBAAmB,SAAQ,wBAAU;CAQjD;AARD,gDAQC;AAMY,QAAA,kBAAkB,GAAG,kBAAS,CAAC,MAAM,CAChD,OAAO,EACP;IACE,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,IAAI,EAAE,qBAAS,CAAC,MAAM;IACtB,MAAM,EAAE,qBAAS,CAAC,OAAO;IACzB,SAAS,EAAE,qBAAS,CAAC,OAAO;IAC5B,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,UAAU,EAAE,qBAAS,CAAC,OAAO;IAC7B,aAAa,EAAE,qBAAS,CAAC,OAAO;IAChC,OAAO,EAAE,qBAAS,CAAC,MAAM;CACe,EAC1C;IACE,SAAS,EAAE,OAAO;IAClB,MAAM,EAAE,UAAU;IAClB,KAAK,EAAE;QACL,YAAY,CAAC,UAAU,EAAE,OAAO;YAC9B,MAAM,QAAQ,GAAG,UAAU,CAAC;YAC5B,QAAQ,CAAC,SAAS,GAAG,IAAA,eAAM,GAAE,CAAC;QAChC,CAAC;KACF;CACF,CACF,CAAC"}