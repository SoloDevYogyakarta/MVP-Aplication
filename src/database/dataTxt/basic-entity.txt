{"id":226,"name":"Generic Granite Table","status":1,"condition":-2,"shortdesc":"Succurro suscipit chirographum. Deripio unus vindico. Quos voluntarius villa constans vero vigilo demo.","main_stock":2,"reserve_stock":2,"user_id":"hbl9pN4evu0OU1W0-bJOS","updatedAt":"2023-10-22T23:13:33.124Z","createdAt":"2023-10-22T23:13:33.124Z","public_id":"nusX-9TrZDOiVWTGhF2Cd"}