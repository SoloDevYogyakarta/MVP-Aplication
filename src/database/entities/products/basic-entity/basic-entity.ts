import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class ProductBasicEntity extends BaseEntity {
  name?: string;
  status?: number;
  condition?: number;
  shortdesc?: string;
  main_stock?: number;
  reserve_stock?: number;
  user_id?: string;
}

export type ProductBasicCreationAttribute = Optional<ProductBasicEntity, 'id'>;
export interface ProductBasicInstance
  extends Model<ProductBasicCreationAttribute, ProductBasicEntity>,
    ProductBasicEntity {}
export const productBasicEntity = sequelize.define<ProductBasicInstance>(
  'BASIC',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    status: DataTypes.INTEGER,
    condition: DataTypes.INTEGER,
    shortdesc: DataTypes.STRING,
    main_stock: DataTypes.INTEGER,
    reserve_stock: DataTypes.INTEGER,
    user_id: DataTypes.STRING,
  } as ModelAttributes<ProductBasicInstance>,
  {
    tableName: 'BASIC',
    schema: 'PRODUCTS',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
