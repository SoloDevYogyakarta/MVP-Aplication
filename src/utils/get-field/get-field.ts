import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
