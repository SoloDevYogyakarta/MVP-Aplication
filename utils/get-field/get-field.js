"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map