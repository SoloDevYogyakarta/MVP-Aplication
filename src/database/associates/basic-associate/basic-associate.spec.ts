import { ModelCtor } from 'sequelize';
import { ProductBasicInstance } from '../../../database/entities/products/basic-entity/basic-entity';
import { getField } from '../../../utils/get-field/get-field';
import {
  productBasicAssociate,
  productBasicAttribute,
  productBasicInclude,
} from './basic-associate';

describe('ProductBasicAssociate', () => {
  let entity: ModelCtor<ProductBasicInstance>;

  beforeEach(() => {
    entity = productBasicAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('findOne with relationship', async () => {
      const { public_id } = getField('basic-http-entity');
      const findOne = await entity.findOne({
        where: { public_id },
        attributes: productBasicAttribute,
        include: productBasicInclude,
      });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll({
        attributes: productBasicAttribute,
        include: productBasicInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
