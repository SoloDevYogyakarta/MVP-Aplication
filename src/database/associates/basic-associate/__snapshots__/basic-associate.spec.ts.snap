// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBasicAssociate render correctly 1`] = `[Function]`;
